src/utils/definitions/general_config.rs

```diff
@@ -41,46 +32,69 @@ pub struct StarknetGeneralConfig {
     validate_max_n_steps: u64,
     min_gas_price: u64,
     constant_gas_price: bool,
+    #[serde(deserialize_with = "felt_from_hex_string")]
     sequencer_address: Felt252,
     tx_commitment_tree_height: u64,
     event_commitment_tree_height: u64,
     cairo_resource_fee_weights: HashMap<String, f32>,
     enforce_l1_handler_fee: bool,
 }
 
-pub static CONFIG: OnceCell<StarknetGeneralConfig> = OnceCell::const_new();
-
-fn starknet_config() -> StarknetGeneralConfig {
-    StarknetGeneralConfig {
-        starknet_os_config: starknet_os_config(),
-        contract_storage_commitment_tree_height: constants::CONTRACT_STATES_COMMITMENT_TREE_HEIGHT,
-        compiled_class_hash_commitment_tree_height:
-            constants::COMPILED_CLASS_HASH_COMMITMENT_TREE_HEIGHT,
-        global_state_commitment_tree_height: constants::CONTRACT_ADDRESS_BITS,
-        invoke_tx_max_n_steps: DEFAULT_TX_MAX_STEPS,
-        validate_max_n_steps: DEFAULT_VALIDATE_MAX_STEPS,
-        min_gas_price: DEFAULT_GAS_PRICE,
-        constant_gas_price: false,
-        sequencer_address: Felt252::new(0), // TODO: Add real value
-        tx_commitment_tree_height: constants::TRANSACTION_COMMITMENT_TREE_HEIGHT,
-        event_commitment_tree_height: constants::EVENT_COMMITMENT_TREE_HEIGHT,
-        cairo_resource_fee_weights: HashMap::default(), // TODO: Add builtins module
-        enforce_l1_handler_fee: DEFAULT_ENFORCE_L1_FEE,
+impl Default for StarknetGeneralConfig {
+    fn default() -> Self {
+        // TODO: handle unwrap
+        Self::try_from(PathBuf::from(DEFAULT_CONFIG_PATH)).unwrap()
     }
 }
 
-impl StarknetGeneralConfig {
-    fn _chain_id(&self) -> Felt252 {
-        self.starknet_os_config.chain_id.clone()
-    }
+impl TryFrom<PathBuf> for StarknetGeneralConfig {
```

Comment:
We could probably have something like `fn load_config<P: AsRef<Path>>(path: P) -> Result<...> {}` too

Reply:
Was originally thinking it would be nice to have configs from multiple sources. i.e.:
- TryFrom -- yaml
- TryFrom -- JSON
- TryFrom -- TOML
^^ Probably YAGNI ^^

Will go with `load_config` for the time being.